pkg/pipeline/stream/base.go

```diff
@@ -344,13 +344,16 @@ func (n *nodeBase) Send(
 	if msg.Ctx == nil {
 		msg.Ctx = ctxutil.ContextWithMessageID(ctx, msg.ID())
 	}
+	// copy context into a local variable, we shouldn't access it anymore after
+	// we send the message to out
+	msgCtx := msg.Ctx
```

Comment:
This solves a race condition where `msg.Ctx` could be changed in another goroutine after it was sent to `out`.